src/main/java/edu/harvard/iq/dataverse/PermissionServiceBean.java

```diff
@@ -963,5 +959,25 @@ public List<Dataverse> findPermittedCollections(DataverseRequest request, Authen
         }
         return null;
     }
+
+    /**
+     * Calculates the complete list of role assignments for a given user on a DvObject.
+     * This includes roles assigned directly to the user and roles inherited from any groups
+     * the user is a member of.
+     *
+     * @param user The authenticated user whose roles are being checked.
+     * @param dvObject The dataverse object to check for role assignments.
+     * @return A List containing all effective RoleAssignments for the user. Never null.
+     */
+    public List<RoleAssignment> getEffectiveRoleAssignments(AuthenticatedUser user, DvObject dvObject) {
+        Stream<RoleAssignment> directAssignments = assignmentsFor(user, dvObject).stream();
+
+        Stream<RoleAssignment> groupAssignments = groupService.groupsFor(user, dvObject)
+                .stream()
+                .flatMap(group -> assignmentsFor(group, dvObject).stream());
+
+        return Stream.concat(directAssignments, groupAssignments)
+                .collect(Collectors.toList());
+    }
```

Comment:
I'm not sure but I feel like this method might already exist somewhere. @sekmiller might know.